ddht/v5_1/alexandria/client.py

```diff
@@ -539,30 +539,49 @@ async def locate(
         content_key: ContentKey,
         request_id: Optional[bytes] = None,
     ) -> Tuple[InboundMessage[LocationsMessage], ...]:
+        stream_locate_ctx = self.stream_locate(
+            node_id, endpoint, content_key=content_key, request_id=request_id,
+        )
+        async with stream_locate_ctx as response_aiter:
+            return tuple([response async for response in response_aiter])
```

Comment:
@njgheorghita here's a loose equivalent to what you're doing where I was able to implement the streaming version and then move the non-streaming version over to just depend on the streaming one.